src/hb-gdi.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2019  Ebrahim Byagowi
+ *
+ *  This is part of HarfBuzz, a text shaping library.
+ *
+ * Permission is hereby granted, without written agreement and without
+ * license or royalty fees, to use, copy, modify, and distribute this
+ * software and its documentation for any purpose, provided that the
+ * above copyright notice and the following two paragraphs appear in
+ * all copies of this software.
+ *
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE TO ANY PARTY FOR
+ * DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES
+ * ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN
+ * IF THE COPYRIGHT HOLDER HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ * THE COPYRIGHT HOLDER SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING,
+ * BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE.  THE SOFTWARE PROVIDED HEREUNDER IS
+ * ON AN "AS IS" BASIS, AND THE COPYRIGHT HOLDER HAS NO OBLIGATION TO
+ * PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+#ifndef HB_GDI_H
+#define HB_GDI_H
+
+#include "hb.h"
+
+#include <windows.h>
+
+HB_BEGIN_DECLS
+
+HB_EXTERN hb_face_t *
+hb_gdi_hfont_face_create (HFONT hfont);
```

Comment:
First.  HFONT has a size, so I feel like it corresponds to `hb_font_t`.  If there's no separate equivalent to hb-face in GDI, it's fine to have both from the HFONT.  That's what we do for `FT_Face`.

Name should be `hb_gdi_font_create()` / `hb_gdi_face_create()`.

Reply:
Reading http://www.winprog.org/tutorial/fonts.html and https://support.microsoft.com/en-ca/help/74299/info-calculating-the-logical-height-and-point-size-of-a-font I am now more familiar with this but am still not sure about how to map lfHeight of LOGFONTW object to our scales, feel free to give me a hand if you like to, or I should learn this by putting some time, that probably will help pango parallel work also, or, lets see what happens there by @fanc999

Reply:
I'm fine moving forward with face for now.  And *not* moving the stuff from Uniscribe here either.  But please consider that in the future.

Reply:
So feel free if it seems fine and you don't have API looking concerns. As I am switching between machines to access mingw for build and test, am considering installing fedora in my main machine.

Reply:
Needs at least rename.  Am fine going in with rename and everything else later.

Reply:
How I missed while I was testing this also :/

Reply:
OK, you are looking at the outdated version I guess

Reply:
Okay.  Comment thread still shows old code.  Np.